src/Features/Core/Portable/ExtractMethod/SelectionResult.cs

```diff
@@ -180,10 +194,35 @@ bool IsConfigureAwaitFalse(SyntaxNode node)
             }
         }
 
+        public DataFlowAnalysis GetDataFlowAnalysis()
+        {
+            return _dataFlowAnalysis ??= ComputeDataFlowAnalysis();
+
+            DataFlowAnalysis ComputeDataFlowAnalysis()
+            {
+                var semanticModel = this.SemanticDocument.SemanticModel;
+                if (this.IsExtractMethodOnExpression)
+                    return semanticModel.AnalyzeDataFlow(this.GetNodeForDataFlowAnalysis());
+
+                var (firstStatement, lastStatement) = this.GetFlowAnalysisNodeRange();
+                return semanticModel.AnalyzeDataFlow(firstStatement, lastStatement);
+            }
+        }
+
+        public ControlFlowAnalysis GetStatementControlFlowAnalysis()
+        {
+            Contract.ThrowIfTrue(IsExtractMethodOnExpression);
+            var (firstStatement, lastStatement) = this.GetFlowAnalysisNodeRange();
```

Comment:
>var (firstStatement, lastStatement) = this.GetFlowAnalysisNodeRange(); [](http://example.com/codeflow?start=12&length=70)

I liked the nested fn approach you took in GetDataFlowAnalysis. Might be nice to do that here too (would avoid the GetFlowAnalysisNodeRange call when _statementControlFlowAnalysis is already computed)

Reply:
good idea.  done.